client/components/platform-checkout/save-user/checkout-page-save-user.js

```diff
@@ -14,27 +14,95 @@ import AdditionalInformation from './additional-information';
 import PhoneNumberInput from 'settings/phone-input';
 import Agreement from './agreement';
 import { getConfig } from 'utils/checkout';
+// eslint-disable-next-line import/no-unresolved
+import { extensionCartUpdate } from '@woocommerce/blocks-checkout';
 
-const CheckoutPageSaveUser = () => {
+const CheckoutPageSaveUser = ( { isBlocksCheckout } ) => {
 	const [ isSaveDetailsChecked, setIsSaveDetailsChecked ] = useState( false );
 	// eslint-disable-next-line no-unused-vars
 	const [ phoneNumber, setPhoneNumber ] = useState( '' );
 	const [ isPhoneValid, onPhoneValidationChange ] = useState( null );
+	const [ userDataSent, setUserDataSent ] = useState( false );
 	const isRegisteredUser = usePlatformCheckoutUser();
-	const {
-		isWCPayChosen,
-		isNewPaymentTokenChosen,
-	} = useSelectedPaymentMethod();
+	const { isWCPayChosen, isNewPaymentTokenChosen } = useSelectedPaymentMethod(
+		isBlocksCheckout
+	);
+
+	const getPhoneFieldValue = () => {
+		let phoneFieldValue = '';
+		if ( isBlocksCheckout ) {
+			phoneFieldValue =
+				document.getElementById( 'phone' )?.value ||
+				document.getElementById( 'shipping-phone' )?.value ||
+				'';
+		} else {
+			// for classic checkout.
+			phoneFieldValue =
+				document.getElementById( 'billing_phone' )?.value || '';
+		}
 
+		setPhoneNumber( phoneFieldValue );
+		return phoneFieldValue;
+	};
+
+	// use extensionCartUpdate for blocks checkout only.
 	useEffect( () => {
+		if ( ! isBlocksCheckout ) {
+			return;
+		}
+
 		const formSubmitButton = document.querySelector(
-			'form.woocommerce-checkout button[type="submit"]'
+			'button.wc-block-components-checkout-place-order-button'
 		);
 
 		if ( ! formSubmitButton ) {
 			return;
 		}
 
+		// send data to extension endpoint when place order button is clicked.
+		const sendRequestToExtension = () => {
+			extensionCartUpdate( {
```

Comment:
Just to confirm, does the checkout request also get triggered from the submit button `click` event?

There's no way to guarantee this is called and processed before the checkout request, right?

Reply:
@bborman22 figured out that the order of the `extensionCartUpdate` and the main checkout request is the cause of why a new user is not created on Blocks checkout sometimes (the issue you and @FangedParakeet both faced before). 
To ensure that the extension data is always set first now the request is sent on the checkbox select/unselect event [22cd5bf](https://github.com/Automattic/woocommerce-payments/pull/4638/commits/22cd5bf8ecb7af7e1a238323b64be3e5994e12e3) :disappointed_relieved: